Source/Schema.NET/ValuesConverter.cs

```diff
@@ -40,8 +34,147 @@ public class ValuesConverter : JsonConverter
             JsonReader reader,
             Type objectType,
             object existingValue,
-            JsonSerializer serializer) =>
-            throw new NotImplementedException();
+            JsonSerializer serializer)
+        {
+            var mainType = UnwrapNullable(objectType);
+
+            object argument = null;
+
+            var tokenType = reader.TokenType;
+            var value = SanitizeReaderValue(reader, tokenType);
+
+            JToken token = JToken.Load(reader);
+            if (mainType.GenericTypeArguments.Length == 1)
+            {
+                var type = UnwrapNullable(mainType.GenericTypeArguments[0]);
+                if (tokenType == JsonToken.StartArray)
+                {
+                    var listType = typeof(List<>).MakeGenericType(type);
+                    if (token.Any(t => !string.IsNullOrEmpty(GetTypeNameFromToken(t))))
+                    {
+                        var list = Activator.CreateInstance(listType);
+                        foreach (var childToken in token.Children())
+                        {
+                            var typeName = GetTypeNameFromToken(childToken);
+                            var builtType = Type.GetType($"Schema.NET.{typeName}");
+                            var child = (Thing)childToken.ToObject(builtType);
+                            listType.GetRuntimeMethod("Add", new[] { type }).Invoke(list, new object[] { child });
+                        }
+
+                        argument = list;
+                    }
+                    else
+                    {
+                        argument = token.ToObject(listType);
+                    }
+                }
+                else if (IsPrimitiveType(type))
+                {
+                    argument = value;
+                }
+                else if (type == typeof(decimal))
+                {
+                    argument = Convert.ToDecimal(value);
+                }
+                else
+                {
+                    if (type.GetTypeInfo().IsEnum)
+                    {
+                        var enumString = token.ToString().Substring("http://schema.org/".Length);
+                        argument = Enum.Parse(type, enumString);
+                    }
+                    else
+                    {
+                        var typeName = GetTypeNameFromToken(token);
+                        if (string.IsNullOrEmpty(typeName))
+                        {
+                            argument = token.ToObject(type);
+                        }
+                        else
+                        {
+                            var builtType = Type.GetType($"Schema.NET.{typeName}");
+                            argument = token.ToObject(builtType);
+                        }
+                    }
+                }
+            }
+            else
+            {
+                foreach (var type in mainType.GenericTypeArguments)
```

Comment:
This is the block that handles the other Values<T....>. Here we need to iterate all the allowed types to see which one is a match.